bliss/simulator/decoder.py

```diff
@@ -14,55 +14,49 @@
 class ImageDecoder(nn.Module):
     def __init__(
         self,
-        n_bands: int,
         pixel_scale: float,
-        psf_params_file: str,
         psf_slen: int,
-        sdss_bands: Tuple[int, ...],
+        sdss_fields: DictConfig,
     ) -> None:
         super().__init__()
-        assert n_bands == 1, "Only 1 band is supported"
 
-        self.n_bands = n_bands
+        self.n_bands = len(sdss_fields["bands"])
         self.pixel_scale = pixel_scale
+        self.psf_slen = psf_slen
+        self.psf_galsim = {}  # Dictionary indexed by (run, camcol, field) tuple
 
-        self.params = None  # psf params from fits file
+        sdss_dir = sdss_fields["dir"]
+        sdss_bands = sdss_fields["bands"]
 
-        assert Path(psf_params_file).suffix == ".fits"
-        assert len(sdss_bands) == n_bands
-        psf_params = self._get_fit_file_psf_params(psf_params_file, sdss_bands)
-        self.params = nn.Parameter(psf_params.clone(), requires_grad=True)
+        for field_params in sdss_fields["field_list"]:
+            run = field_params["run"]
+            camcol = field_params["camcol"]
+            fields = field_params["fields"]
 
-        self.psf_slen = psf_slen
+            for field in fields:
```

Comment:
L36-43 results in the construction of the `self.psf_galsim` dictionary, which is indexed by `(<run>, <camcol>, <field>)` and returns the PSFs for that particular field. Particularly, only the requested PSFs are generated based on the bands specified in `sdss_bands`.